metaflow/client/core.py

```diff
@@ -680,13 +685,33 @@ def data(self):
             obj = pickle.load(f)
             return obj
 
-    # TODO add
-    # @property
-    # def size(self)
+    @property
+    def size(self):
+        """
+        Unpickled size of the data contained in this artifact.
 
-    # TODO add
-    # @property
-    # def type(self)
+        Returns
+        -------
+        size
+            Object size contained in this artifact
+        """
+        ds_type = self._object['ds_type']
+        sha = self._object['sha']
+        with filecache.get_data(ds_type, self.path_components[0], sha) as f:
```

Comment:
I am not a big fan of having to unpickle and then throw away the result just to get the size (and doing it every time). A potentially better approach would be to store the size information along with the sha, ds_type, etc and return that directly.

Reply:
@romain-intel I couldn't agree with the approach you mentioned. The sha, ds_type, etc are stored in json. Storing the size information into json is unreasonable. This should calculate the size by `get_data`.

Reply:
@SteNicholas Why do you think storing the size information is unreasonable?

Reply:
@savingoyal In my option, the sha, ds_type etc. information is stored in json file. And the size couldn't be defined the value in json.

Reply:
Why not?

Reply:
@savingoyal Do you mean that the size should be defined the value in json? I will modify this by storing the size in json.

Reply:
yes, indeed.

Reply:
@savingoyal I have already modified the size storing in info. Please review this again.

Reply:
Thanks, I will take a look in the morning.